module2/projects/hang-in-there-api/requirements.md

```diff
@@ -0,0 +1,579 @@
+---
+layout: page
+title: Hang In There API Requirements
+length: 1 week
+tags:
+type: project
+---
+<style>
+summary:hover {
+  background-color: #bbe5fa;
+}
+</style>
+
+
+_[Back to Hang In There API Home](./index)_
+## 1. Setup
+
+1. Create a Rails API project called `hang_in_there_api` with the command `rails new hang_in_there_api -T -d="postgresql" --api`
+
+2. Add your gems to your `development, test` section in the Gemfile:
+    * rspec-rails
+    * pry
+    * simplecov
+
+3. Run the following commands to get started: 
+    ```bash
+    $ cd hang_in_there_api
+    $ bundle install
+    $ rails db:create
+    $ rails g rspec:install
+    ```
+
+4. Your database will have one table: `posters`. Use migrations to create this table with the following attributes:
+
+    * `id` as integer
+    * `name` as string
+    * `description` as string
+    * `price` as float
+    * `year` as integer
+    * `vintage` as boolean
+    * `img_url` as string
+    * `created_at` as timestamp
+    * `updated_at` as timestamp
+
+5. It will make it easier to test as you develop if you create some seeds in your `seeds.rb` file. You can use the following pattern to create seed data.
+  ```ruby
+    Poster.create(name: "REGRET",
+                  description: "Hard work rarely pays off.",
+                  price: 89.00,
+                  year: 2018,
+                  vintage: true,
+                  img_url:  "https://plus.unsplash.com/premium_photo-1661293818249-fddbddf07a5d", )
+  ```
+
+6. Finally, commit your setup steps and push to a new repo. Share that new repo with your project partner(s). Be sure to add them as a collaborator.
+
+---
+
+## 2. API Endpoints, general definitions
+
+You will need to expose the data through a multitude of API endpoints. All of your endpoints should follow these technical expectations:
+
+* All endpoints will expect to return JSON data only
+* All endpoints should be exposed under an `api` and version (`v1`) namespace (e.g. `/api/v1/posters`)
+* API will be compliant to the [JSON API spec](https://jsonapi.org/) and match our requirements below precisely
+* Controller actions should be limited to only the standard Rails actions and follow good RESTful convention.
+
+
+You will need to expose the following RESTful API endpoints for the following:
+
+<details><summary><h3>1. Fetch all Posters</h3></summary>
```

Comment:
Are you open to using the dropdown formatting we've used in lessons for these dropdowns?  I think it makes it easier to tell at a glance that its a drop down and will keep things more consistent?

The syntax is easy!

```js 
<section class="dropdown">
### 1. Fetch all Posters

These "index" endpoints should:
....
</section>

```

![image](https://github.com/user-attachments/assets/10bf4364-376e-46ea-bc8c-87bdda6191d3)

Reply:
I AM open to it! I was struggling with the syntax. Thank you!!